File: CareConnect/CareConnect.Models/SearchObjects/EmployeeAvailabilityAdditionalData.cs
using CareConnect.Models.SearchObjects;

namespace CareConnect.Models.SearchObjects
{
    public class EmployeeAvailabilityAdditionalData : BaseAdditionalSearchRequestData
    {
        public bool? IsEmployeeIncluded { get; set; }

        public bool? IsServiceIncluded { get; set; }

    }
}

File: UI/careconnect_admin/lib/models/search_objects/employee_availability_additional_data.dart
import 'package:json_annotation/json_annotation.dart';

part 'employee_availability_additional_data.g.dart';

@JsonSerializable(explicitToJson: true)
class EmployeeAvailabilityAdditionalData {
  bool? isEmployeeIncluded;
  bool? isServiceIncluded;

  EmployeeAvailabilityAdditionalData({
    this.isEmployeeIncluded,
    this.isServiceIncluded,
  });

  factory EmployeeAvailabilityAdditionalData.fromJson(
    Map<String, dynamic> json,
  ) => _$EmployeeAvailabilityAdditionalDataFromJson(json);

  Map<String, dynamic> toJson() =>
      _$EmployeeAvailabilityAdditionalDataToJson(this);
}
